### ISSUE TITLE
Azure Web App Deployment fails for non-master branches

### ISSUE THREAD

**Issue by kaplas** [travis-ci/travis-ci#5468]
Azure Web App Deployment for a (Node.js) project fails for non-`master` branches with this error message:
```
error: src refspec master does not match any.
```

I had to add this pre-build-script for the deployment to work regardless of the current branch:
```yml
before_deploy:
- echo "Adding refs/heads/master for Azure deployment to work"
- git checkout --no-track -b master || true
- git show-ref
```

I'm now a git expert in anyway (actually, far far away from being one), but the problem seems to be that git is trying to push to the Azure git repository from the local `master` branch, which will fail, because `refs/heads/master` is missing directly after when a repo is cloned with `--branch=<branch_name>`.

In other words, I imagine, that your current deploy script is now doing something like
```
git push -f azure master
```
where it actually should be executing
```
git push -f azure <branch_name>:master
```

_Ps. I could perhaps actually submitted a PR for this, but frankly, I did not know what is the right repository to look for the deploy scripts._

**Comment by BanzaiMan** [comment]
https://github.com/travis-ci/dpl/blob/master/lib/dpl/provider/azure_webapps.rb is the implementation.

**Comment by kaplas** [comment]
Thanx! I will try to create a PR :)
